tests/unit/test_message_queue.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ *  Copyright (c) 2016, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test_util.h"
+#include <openthread.h>
+#include <common/debug.hpp>
+#include <common/message.hpp>
+#include <string.h>
+#include <stdarg.h>
+
+#define kNumMessages      5
+
+// This function verifies the content of the message to matches the passed in messages
+void VerifyMessageQueueContent(Thread::MessageQueue &aMessageQueue, int aExpectedLength, ...)
+{
+    va_list args;
+    Thread::Message *message;
+    Thread::Message *msgArg;
+
+    va_start(args, aExpectedLength);
+
+    if (aExpectedLength == 0)
+    {
+        message = aMessageQueue.GetHead();
+        VerifyOrQuit(message == NULL, "MessageQueue is not empty when expected len is zero.");
+    }
+    else
+    {
+        for (message = aMessageQueue.GetHead(); message != NULL; message = message->GetNext())
+        {
+            VerifyOrQuit(aExpectedLength != 0, "MessageQueue contains more entries than expected");
+
+            msgArg = va_arg(args, Thread::Message *);
+            VerifyOrQuit(msgArg == message, "MessageQueue content does not match what is expected.");
+
+            aExpectedLength--;
+        }
+
+        VerifyOrQuit(aExpectedLength == 0, "MessageQueue contains less entries than expected");
+    }
+
+    va_end(args);
+}
+
+void TestMessageQueue(void)
```

Comment:
Yes. Good idea. Can you please help add this later (in a separate commit/PR)? Thanks. 
I am not very familiar with the Windows make/project files and unit test code.